Spyking circus crash during matching

Hello, I have been managing to crash Spyking circus with the following error. It seems to happen randomly (I have tried running it a number of times, and it will crash with the same error at different probes). I am sorting a Utah array and run the sorter using the ```run_sorter_by_property```  function. This recording does have several instances of drop out, could this be the underlying problem? If so, might there be a parameter I could change to help? Thanks for your time!


```
SpikeSortingError: Spike sorting error trace:
Traceback (most recent call last):
  File "/media/paul/storage/pdev/test/spikeinterface/src/spikeinterface/sorters/basesorter.py", line 270, in run_from_folder
    SorterClass._run_from_folder(sorter_output_folder, sorter_params, verbose)
  File "/media/paul/storage/pdev/test/spikeinterface/src/spikeinterface/sorters/internal/spyking_circus2.py", line 272, in _run_from_folder
    labels, peak_labels = find_cluster_from_peaks(
                          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/paul/storage/pdev/test/spikeinterface/src/spikeinterface/sortingcomponents/clustering/main.py", line 44, in find_cluster_from_peaks
    outputs = method_class.main_function(recording, peaks, params, job_kwargs=job_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/paul/storage/pdev/test/spikeinterface/src/spikeinterface/sortingcomponents/clustering/random_projections.py", line 182, in main_function
    labels, peak_labels = remove_duplicates_via_matching(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/paul/storage/pdev/test/spikeinterface/src/spikeinterface/sortingcomponents/clustering/clustering_tools.py", line 666, in remove_duplicates_via_matching
    similar_templates = detect_mixtures(
                        ^^^^^^^^^^^^^^^^
  File "/media/paul/storage/pdev/test/spikeinterface/src/spikeinterface/sortingcomponents/clustering/clustering_tools.py", line 655, in detect_mixtures
    del recording, sub_recording, local_params, templates
                   ^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'sub_recording' where it is not associated with a value

Spike sorting failed. You can inspect the runtime trace in /media/paul/storage/Data/test/processed/results_sc/6/spikeinterface_log.json.
```

Thanks for checking! I am installing spikeinterface with anaconda. So I deleted the environment, installed all the dependencies basically using the yml file from the "installation tips" section, but excluding spikeinterface. Then I clone the repository and installed spikeinterface from the code. Then I use ```git status``` to check that code is ok. Then I use ```gh pr checkout 3820``` 

You comment made me realize that I still needed to reinstall the updated code, so I uninstalled spikeinterface using pip, and then reinstalled. The version it gives me 0.102.2, which I thought we needed 0.102.1? But rerunning my code again produced a different error, so maybe I finally figured out how to install it correctly!

```
SpikeSortingError: Spike sorting error trace:
Traceback (most recent call last):
  File "/home/paul/anaconda3/envs/sind/lib/python3.11/site-packages/spikeinterface-0.102.2-py3.11.egg/spikeinterface/sorters/basesorter.py", line 270, in run_from_folder
    SorterClass._run_from_folder(sorter_output_folder, sorter_params, verbose)
  File "/home/paul/anaconda3/envs/sind/lib/python3.11/site-packages/spikeinterface-0.102.2-py3.11.egg/spikeinterface/sorters/internal/spyking_circus2.py", line 293, in _run_from_folder
    templates_array = estimate_templates(
                      ^^^^^^^^^^^^^^^^^^^
  File "/home/paul/anaconda3/envs/sind/lib/python3.11/site-packages/spikeinterface-0.102.2-py3.11.egg/spikeinterface/core/waveform_tools.py", line 754, in estimate_templates
    templates_array = estimate_templates_with_accumulator(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/paul/anaconda3/envs/sind/lib/python3.11/site-packages/spikeinterface-0.102.2-py3.11.egg/spikeinterface/core/waveform_tools.py", line 828, in estimate_templates_with_accumulator
    assert spikes.size > 0, "estimate_templates() need non empty sorting"
           ^^^^^^^^^^^^^^^
AssertionError: estimate_templates() need non empty sorting
```

--- a/src/spikeinterface/sortingcomponents/clustering/clustering_tools.py
+++ b/src/spikeinterface/sortingcomponents/clustering/clustering_tools.py
@@ -590,10 +590,10 @@ def detect_mixtures(templates, method_kwargs={}, job_kwargs={}, tmp_folder=None,
     local_job_kargs = {"n_jobs": 1, "progress_bar": False}
 
     DEBUG = False
+    sub_recording = None  # we only do a sub recording if we have data to iterate through
     while keep_searching:
 
         keep_searching = False
-        sub_recording = None  # we only do a sub recording if we have data to iterate through
 
         for i in list(set(range(nb_templates)).difference(ignore_inds)):
 

--- a/src/spikeinterface/sortingcomponents/clustering/clustering_tools.py
+++ b/src/spikeinterface/sortingcomponents/clustering/clustering_tools.py
@@ -593,6 +593,7 @@ def detect_mixtures(templates, method_kwargs={}, job_kwargs={}, tmp_folder=None,
     while keep_searching:
 
         keep_searching = False
+        sub_recording = None  # we only do a sub recording if we have data to iterate through
 
         for i in list(set(range(nb_templates)).difference(ignore_inds)):
 

